### PR TITLE
Pull event name transformation into engine.

### LINKED ISSUE #390: Event Optimization
Ok, did a quick profile run (100 frames of `ppb_mutant.index`), and the results point to the event system:

```
   ncalls  tottime  percall  cumtime  percall filename:lineno(function)
 22043200   59.004    0.000   77.684    0.000 {method 'sub' of 're.Pattern' objects}
 11021600   16.613    0.000  158.867    0.000 eventlib.py:49(__event__)
 22043200   16.042    0.000   16.042    0.000 re.py:307(_subx)
 11021607   13.460    0.000   28.815    0.000 __init__.py:1216(getLogger)
 11021600    9.267    0.000   88.802    0.000 eventlib.py:16(camel_to_snake)
    11980    7.999    0.001  172.005    0.014 engine.py:125(publish)
 11116344    6.146    0.000    6.150    0.000 {built-in method builtins.getattr}
 11021607    5.630    0.000   34.445    0.000 __init__.py:1928(getLogger)
 11021617    4.066    0.000    7.430    0.000 __init__.py:212(_acquireLock)
   276525    4.050    0.000    4.134    0.000 _sdl_utils.py:25(sdl_call)
```

### PR DESCRIPTION
**Pull request by pathunstrom**
Closes #390

### CODE REVIEW

File: ppb/engine.py [hunk 1]
```diff
@@ -127,17 +141,26 @@ def publish(self):
         scene = self.current_scene
         event.scene = scene
         extensions = chain(self.event_extensions[type(event)], self.event_extensions[...])
+
+        # Hydrating extensions.
```
Comment (astronouth7303) [inline ppb/engine.py]: Odd phrasing?

File: tests/test_events.py [hunk 2]
```diff
@@ -1,62 +1,8 @@
...
-    fire_event = lambda: None
-
-    e = Eventable()
-
-    e.__event__(bag, fire_event)
-    assert bag is passed_bag
-    assert fire_event is passed_fire
-
-
-def test_event_mixin_with_bad_signature():
```
Comment (astronouth7303) [inline tests/test_events.py]: Not a fan of losing this test, but I also can't think of a good way to handle it.

Comment (pathunstrom) [reply tests/test_events.py]: Yup, I think I need to puzzle on how to get it. It's a lot _heavier_ of a test now.

**Pull request merged**